src/lib/y2partitioner/dialogs/btrfs_subvolume.rb

```diff
@@ -266,6 +270,201 @@ def help
+            "system. Btrfs quotas can be enabled or disabled editing the file system " \
+            "from the Btrfs section of the Partitioner.</p>"
+          )
+        end
+
+        # Disables the widget if quotas are not enabled
+        #
+        # @see #disable_widgets
+        def init
+          disable_widgets unless quota?
+        end
+
+        # @macro seeAbstractWidget
+        def value
+          if quota? && check_box_widget.value
+            size_widget.value
+          else
+            Y2Storage::DiskSize.unlimited
+          end
+        end
+
+        # @macro seeAbstractWidget
+        def store
+          @controller.subvolume_referenced_limit = value
+        end
+
+        # @macro seeAbstractWidget
+        def validate
+          return true if value
+
+          Yast::Popup.Error(_("The size entered is invalid."))
+          Yast::UI.SetFocus(Id(size_widget.widget_id))
+          false
+        end
+
+        private
+
+        # Widget for the checkbox used to add/remove the quota
+        #
+        # @return [SubvolumeRferLimitCheckBox]
+        def check_box_widget
+          @check_box_widget ||= SubvolumeRferLimitCheckBox.new(initial_check_box, size_widget)
+        end
+
+        # Widget to introduce the size of the quota
+        #
+        # @return [SubvolumeRferLimitSize]
+        def size_widget
+          @size_widget ||= SubvolumeRferLimitSize.new(initial_size)
+        end
+
+        # Disables the sub-widgets
+        #
+        # To be used when quotas are not enabled
+        def disable_widgets
+          check_box_widget.disable
+          size_widget.disable
+        end
+
+        # Whether quotas are enabled for the file system
+        #
+        # @return [Boolean]
+        def quota?
+          @controller.quota?
+        end
+
+        # Initial value for {#size_widget}
+        #
+        # @return [Y2Storage::DiskSize]
+        def initial_size
+          initial_limit = @controller.subvolume_referenced_limit
+          if !initial_limit || initial_limit.unlimited?
+            @controller.fallback_referenced_limit
+          else
+            @controller.subvolume_referenced_limit
+          end
+        end
+
+        # Initial value for {#check_box_widget}
+        #
+        # @return [Boolean]
+        def initial_check_box
+          initial_limit = @controller.subvolume_referenced_limit
+          return false unless initial_limit
+
+          !initial_limit.unlimited?
+        end
+      end
+
+      # Sub-widget of SubvolumeRferLimit used to enter the size if the
+      # limit is set
+      class SubvolumeRferLimitSize < CWM::InputField
+        include SizeParser
+
+        # Constructor
+        #
+        # @param initial [Y2Storage::DiskSize] initial value
+        def initialize(initial)
+          textdomain "storage"
+          @initial = initial
+        end
+
+        # @macro seeAbstractWidget
+        def label
```

Comment:
What do you think about adding some help in this widget to explain the referenced size?

Reply:
Done.